src/main/java/com/worksap/fig/lang/Seq.java

```diff
@@ -404,4 +431,189 @@ default int findLastIndex(Predicate<T> condition) {
     default Seq<T> subSeq(int fromIndex, int toIndex) {
         return new SeqImpl<>(this.subList(fromIndex, toIndex));
     }
+
+    /**
+     * Remove elements which satisfy the condition, resulting a new seq without changing the original one.
+     *
+     * @return The new seq after the change
+     */
+    default Seq<T> removeIf(BiPredicate<T, Integer> condition) {
+        Objects.requireNonNull(condition);
+        Seq<T> copy = new SeqImpl<>();
+        this.forEachWithIndex((e, i) -> {
+            if (!condition.test(e, i))
+                copy.add(e);
+        });
+        return copy;
+    }
+
+    /**
+     * Remove elements which satisfy the condition at the original seq.
+     *
+     * @return The seq itself after the change
+     */
+    default Seq<T> removeIf$(BiPredicate<T, Integer> condition) {
```

Comment:
Consider to use `reject`. Ruby has the style to make many names on the same function. But I think we don't need to follow that in Java. So I think to make the best name is very important